src/externalized/tactical/MapItemReplacementModel.cc

```diff
@@ -1,27 +1,28 @@
 #include "MapItemReplacementModel.h"
 
-static const ItemModel* GetItemFromValue(const rapidjson::Value& val, const ItemSystem* items)
+static const ItemModel* GetItemFromValue(const JsonValue& val, const ItemSystem* items)
 {
-	if (val.IsString())
-	{
-		return items->getItemByName(val.GetString());
-	}
-	if (val.IsInt())
-	{
-		return items->getItem(val.GetInt());
-	}
+	try {
+		auto str = val.toString();
+		return items->getItemByName(str);
+	} catch (...) {}
+		try {
```

Comment:
Either the indentation is weird here or this does not do what you meant to do.

Reply:
Good catch. This is one of two places I still need to refactor, as they should not use `try {} catch {}`. I was just too lazy to define `.isXXX` methods on `JsonValue`.